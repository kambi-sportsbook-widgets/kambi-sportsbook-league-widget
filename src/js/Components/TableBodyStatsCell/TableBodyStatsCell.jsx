import React, { PropTypes } from 'react';
import { translationModule } from 'kambi-widget-core-library';
import styles from './TableBodyStatsCell.scss';

const t = translationModule.getTranslation.bind(translationModule);

const TableBodyStatsCell = ({ column, value }) => {
   return (
      <td
         title={t(column.name)}
         className={[styles.general, column.className].join(' ')}
      >
         {value}
      </td>
   );
};

TableBodyStatsCell.propTypes = {
   /**
    * Statistics column definition
    */
   column: PropTypes.shape({
      name: PropTypes.string.isRequired,
      className: PropTypes.string.isRequired
   }).isRequired,

   /**
    * Statistics value
    */
   value: PropTypes.number
};

export default TableBodyStatsCell;
