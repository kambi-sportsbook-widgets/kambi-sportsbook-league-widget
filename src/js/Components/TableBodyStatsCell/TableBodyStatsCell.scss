@import '../TableBodyShared';

.general {
   @extend .cell;

   text-align: center;
   width: 5.5%;

   &.points {
      font-weight: bold;
   }
}
