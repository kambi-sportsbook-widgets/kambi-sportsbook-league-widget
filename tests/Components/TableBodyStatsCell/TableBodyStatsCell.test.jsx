/* eslint-env jest */
import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import TableBodyStatsCell from '../../../src/js/Components/TableBodyStatsCell/TableBodyStatsCell';

let renderer;

jest.mock('kambi-widget-core-library', () => ({
   translationModule: {
      getTranslation: (key) => `Translated: "${key}"`
   }
}));

describe('TableBodyStatsCell DOM rendering', () => {

   beforeEach(() => {
      renderer = new ReactShallowRenderer();
   });

   it('renders correctly without value', () => {
      expect(renderer.render(
         <TableBodyStatsCell column={{name: 'Test stat name', className: 'test-stat-name'}} />
      )).toMatchSnapshot();
   });

   it('renders correctly with value', () => {
      expect(renderer.render(
         <TableBodyStatsCell column={{name: 'Test stat name', className: 'test-stat-name'}}>5</TableBodyStatsCell>
      )).toMatchSnapshot();
   });

});
