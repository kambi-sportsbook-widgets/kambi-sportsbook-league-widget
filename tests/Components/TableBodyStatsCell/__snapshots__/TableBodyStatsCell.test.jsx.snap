// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBodyStatsCell DOM rendering renders correctly with value 1`] = `
<td
  className="general test-stat-name"
  title="Translated: \\"Test stat name\\""
/>
`;

exports[`TableBodyStatsCell DOM rendering renders correctly without value 1`] = `
<td
  className="general test-stat-name"
  title="Translated: \\"Test stat name\\""
/>
`;
